trie/child_storage.go

```diff
@@ -0,0 +1,96 @@
+package trie
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/common"
+)
+
+// ChildStorageKeyPrefix is the prefix for all child storage keys
+var ChildStorageKeyPrefix = []byte(":child_storage:")
+
+// PutChild inserts a child trie into the main trie at key :child_storage:[keyToChild]
+func (t *Trie) PutChild(keyToChild []byte, child *Trie) error {
+	childHash, err := child.Hash()
+	if err != nil {
+		return err
+	}
+
+	key := append(ChildStorageKeyPrefix, keyToChild[:]...)
+	exists, err := t.Get(key)
+	if err != nil {
+		return err
+	}
+
+	if exists != nil {
+		err = t.Delete(key)
+		if err != nil {
+			return err
+		}
+	}
+
+	value := [32]byte(childHash)
+
+	err = t.Put(key, value[:])
+	if err != nil {
+		return err
+	}
+
+	t.children[common.Hash(childHash)] = child
```

Comment:
Already a `common.Hash`. This is repeated throughout this file

Reply:
fixed